Marlin/src/inc/SanityCheck.h

```diff
@@ -4380,8 +4380,16 @@ static_assert(_PLUS_TEST(3), "DEFAULT_MAX_ACCELERATION values must be positive."
   #elif DISABLED(FTM_UNIFIED_BWS)
     #error "FT_MOTION requires FTM_UNIFIED_BWS to be enabled because FBS is not yet implemented."
   #endif
+  #if !HAS_X_AXIS && !HAS_Y_AXIS
+    #warning "FT_MOTION shaping needs an X or an Y axis."
+  #else
+    #define FTM_SHAPING
+  #endif
   #if !HAS_X_AXIS
-    static_assert(FTM_DEFAULT_SHAPER_X != ftMotionShaper_NONE, "Without any linear axes FTM_DEFAULT_SHAPER_X must be ftMotionShaper_NONE.");
+    static_assert(FTM_DEFAULT_SHAPER_X != ftMotionShaper_NONE, "Without X axis FTM_DEFAULT_SHAPER_X must be ftMotionShaper_NONE.");
+  #endif
+  #if !HAS_Y_AXIS
+    static_assert(FTM_DEFAULT_SHAPER_Y != ftMotionShaper_NONE, "Without Y axis FTM_DEFAULT_SHAPER_Y must be ftMotionShaper_NONE.");
```

Comment:
There's no such thing as a Y axis without an X axis, so there is no need to change this.